home work 24/script.js

```diff
@@ -0,0 +1,52 @@
+const productsApiUrl =
+  "https://api.escuelajs.co/api/v1/products/?offset=0&limit=10";
+const productsContainer = document.querySelector(".products-container");
+const productDetails = document.querySelector(".product-details");
+
+function fetchProducts() {
+  fetch(productsApiUrl)
+    .then((response) => response.json())
+    .then((data) => {
+      productsContainer.innerHTML = "";
+      data.forEach((product) => {
+        const productCard = createProductCard(product);
+        productsContainer.appendChild(productCard);
+      });
+    })
+    .catch((error) => console.error(error));
+}
+
+function createProductCard(product) {
+  const productCard = document.createElement("div");
+  productCard.classList.add("product-card");
+  productCard.innerHTML = `
+    <img class=".product-img" src="${product.images[0]}" alt="${product.title}">
+    <div class="product-name">${product.title}</div>
+    <div class="product-price">$${product.price.toFixed(2)}</div>
+    <button class="add-to-cart">Add to Cart</button>
+  `;
+
+  productsContainer.addEventListener("click", (event) => {
```

Comment:
You call `createProductCard` in a loop hence you will add a listener per iteration. And as you know `addEventListener` is able to add a few listeners for the same event even. You need to separate this code